### DISCUSSION TITLE
ATC - synchronize with sensor

### DISCUSSION THREAD

**Discussion by MARIOBASZ** [vlachoudis/bCNC#1695] [Q&A]
I need to synchronize the reading of a sensor and according to the result, and within a maximum established time, if the value is correct, continue with the automatic change of tools.
However, the sensor readings and timeout correspond to when Python sends the commands, which is many seconds before the machine must interpret this data.
Self-taught and with limited knowledge.
What I'm looking for is this:
What is the way to synchronize, or wait until the buffer has sent all the data before starting the tool change?

**Comment by Harvie** [comment]
What ATC do you use?

**Comment by MARIOBASZ** [comment]
Hi Thomas, how are you?
It is an own development, similar to others (with some differences) that you find on the web, without using pneumatics: it is about tightening or loosening the spindle nut. I have the data from the sensor that tells me the tightening torque. The information from that sensor is what I need to sync when the spindle is in the right location.
I started with this project three years ago, I think it's almost ready. I intuit that for a traditional ATC it is also necessary to synchronize information and real position.
The fact that it is going to be a commercial item, is it against the spirit of requesting help on this site? If so, I apologize and withdraw the request for help.

**Reply by Harvie** [reply]
> It is an own development, similar to others (with some differences) that you find on the web, without using pneumatics

Something like this?

https://hackaday.io/project/12308-xatc-extreme-simple-automatic-tool-changer

> it is going to be a commercial item, is it against the spirit of requesting help on this site?

not at all. i like the idea of ATC for small hobby machines.

**Comment by MARIOBASZ** [comment]
Yes, like XATC, and
https://www.youtube.com/watch?v=NdT5LD6asIc&t=101s
https://youtu.be/qKVBNwFpyJM

**Comment by MARIOBASZ** [comment]
I think I've found a way to fix it.
